x-pack/plugin/ent-search/src/test/java/org/elasticsearch/xpack/application/search/action/PutSearchApplicationActionResponseBWCSerializingTests.java

```diff
@@ -30,4 +31,12 @@ protected PutSearchApplicationAction.Response createTestInstance() {
     protected PutSearchApplicationAction.Response mutateInstance(PutSearchApplicationAction.Response instance) throws IOException {
         return randomValueOtherThan(instance, this::createTestInstance);
     }
+
+    @Override
+    protected PutSearchApplicationAction.Response mutateInstanceForVersion(
```

Comment:
we have to implement this because we are now inheriting from `AbstractBWCWireSerializationTestCase`.

this is not related to your PR, but looking at the other (~70) implementation of this method most of them are the same where we just do `return instance`.

Is there a reason why this is an abstract method?

https://github.com/elastic/elasticsearch/blob/a2d47998f3d8ed3f37161441c4416bae436eb522/x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/AbstractBWCSerializationTestCase.java#L30

Reply:
That's a really good question.

My guess is that the assumption was, these tests wouldn't be written unless we had backwards compatibility changes to address. However, if we're going to be proactively including them (which I think is a good idea) then it would make sense to have a sane default to return the instance.

Reply:
I guess the intention is to make us _think_ about that. If we're doing a BWC test, then _something_ could have changed between version serializations. So probably it's _let's not forget_?

Reply:
(@kderusso - sorry again, I know this is not related to your PR and it's not a change I propose doing here)
I don't know if there's a clear intention behind making this method abstract or if all implementations just follow a pattern that's already in place.
This method was added as part of https://github.com/elastic/elasticsearch/pull/69534 by @astefan.

@astefan - do you think it would make sense to provide a sane default here instead of having this be an abstract method and forcing an implementation in each subclass?